keps/sig-auth/2579-psp-replacement/README.md

```diff
@@ -0,0 +1,1086 @@
+permissions so they are not equivalent to root-on-node (or cluster).
+
+#### Requirements
+
+1. Validating only (i.e. no changing pods to make them comply with policy)
+2. Safe to enable in new AND upgraded clusters
+    - Dryrun policy changes and/or Audit-only mode
+3. Built-in in-tree controller
+4. Capable of supporting Windows in the future, if not in the initial release
+    - Don’t automatically break windows pods
+5. Must be responsive to Pod API evolution across versions
+6. (fuzzy) Easy to use, don’t need to be a kubernetes/security/linux expert to meet the basic objective
+7. Extensible: should work with custom policy implementations without whole-sale replacement
+    - Enable custom policies keyed off of RuntimeClassNames
+
+Nice to have:
+
+1. Exceptions or policy bindings by requesting user
+2. (fuzzy) Windows support in the initial release
+3. Admission controller is enabled by default in beta or GA phase
+4. Provide an easy migration path from PodSecurityPolicy
+5. Enforcement on pod-controller resources (i.e. things embedding a PodTemplate)
+
+### Non-Goals
+
+1. Provide a configurable policy mechanism that meets the needs of all use-cases
+2. Limit privilege escalation and other attacks beyond the pod to host boundary
+   (e.g. policies on services, secrets, etc.)
+3. Feature parity with PodSecurityPolicy. In particular, support for providing default values or any
+   other mutations will not be included.
+
+## Proposal
+
+The three profile levels (privileged, baseline, restricted) of the [Pod Security Standards] will
+be hardcoded into the new admission plugin. Changes to the standards will be tied to the Kubernetes
+version that they were introduced in.
+
+### API
+
+Policy application is controlled based on labels on the namespace. The following labels are supported:
+```
+pod-security.kubernetes.io/enforce: <policy level>
+pod-security.kubernetes.io/enforce-version: <policy version>
+pod-security.kubernetes.io/audit: <policy level>
+pod-security.kubernetes.io/audit-version: <policy version>
+pod-security.kubernetes.io/warn: <policy level>
+pod-security.kubernetes.io/warn-version: <policy version>
+```
+
+**Enforce:** Pods meeting the requirements of the enforced level are allowed. Violations are rejected
+in admission.
+
+**Audit:** Pods and [templated pods] meeting the requirements of the audit policy level are ignored.
+Violations are recorded in a `pod-security.kubernetes.io/audit-violations: <violation>` [audit
+annotation](#audit-annotations) on the audit event for the request. Audit annotations will **not**
+be applied to the pod objects themselves, as doing so would violate the non-mutating requirement.
+
+**Warn:** Pods and [templated pods] meeting the requirements of the warn level are ignored.
+Violations are returned in a user-facing warning message. Warn & audit modes are independent; if the
+functionality of both is desired, then both labels must be set.
```

Comment:
Why are they independent? In what scenarios would I want one but definitely not want the other?